File: backend/internal/consts/service_types.go
package consts

// 服务类型常量定义
const (
	ServiceTypeMeal       = "meal"       // 送餐服务
	ServiceTypeMedical    = "medical"    // 就医陪护
	ServiceTypeCare       = "care"       // 日常照护
	ServiceTypeRepair     = "repair"     // 居家维修
	ServiceTypeCleaning   = "cleaning"   // 家政保洁
	ServiceTypeCompany    = "company"    // 陪伴聊天
	ServiceTypeEmergency  = "emergency"  // 紧急救助
	ServiceTypeShopping   = "shopping"   // 代买代购
	ServiceTypeTransport  = "transport"  // 出行接送
	ServiceTypeRehab      = "rehab"      // 康复理疗
	ServiceTypePsychology = "psychology" // 心理慰藉
	ServiceTypeLegalAid   = "legal_aid"  // 法律援助
	ServiceTypeOther      = "other"      // 其他服务
)

// ServiceTypes 所有服务类型列表
var ServiceTypes = []string{
	ServiceTypeMeal,
	ServiceTypeMedical,
	ServiceTypeCare,
	ServiceTypeRepair,
	ServiceTypeCleaning,
	ServiceTypeCompany,
	ServiceTypeEmergency,
	ServiceTypeShopping,
	ServiceTypeTransport,
	ServiceTypeRehab,
	ServiceTypePsychology,
	ServiceTypeLegalAid,
	ServiceTypeOther,
}

// ServiceTypeNames 服务类型显示名称映射
var ServiceTypeNames = map[string]string{
	ServiceTypeMeal:       "送餐服务",
	ServiceTypeMedical:    "就医陪护",
	ServiceTypeCare:       "日常照护",
	ServiceTypeRepair:     "居家维修",
	ServiceTypeCleaning:   "家政保洁",
	ServiceTypeCompany:    "陪伴聊天",
	ServiceTypeEmergency:  "紧急救助",
	ServiceTypeShopping:   "代买代购",
	ServiceTypeTransport:  "出行接送",
	ServiceTypeRehab:      "康复理疗",
	ServiceTypePsychology: "心理慰藉",
	ServiceTypeLegalAid:   "法律援助",
	ServiceTypeOther:      "其他服务",
}

// IsValidServiceType 验证服务类型是否合法
func IsValidServiceType(serviceType string) bool {
	for _, t := range ServiceTypes {
		if t == serviceType {
			return true
		}
	}
	return false
}

// GetServiceTypeName 获取服务类型显示名称
func GetServiceTypeName(serviceType string) string {
	if name, ok := ServiceTypeNames[serviceType]; ok {
		return name
	}
	return "未知服务"
}

File: frontend/c-end/src/config/serviceTypes.ts
// 服务类型配置（与后端 consts/service_types.go 保持一致）

export interface ServiceType {
  key: string
  name: string
  icon: string
  description?: string
}

// 后端支持的服务类型枚举
export type ServiceTypeKey =
  | 'meal'
  | 'medical'
  | 'care'
  | 'repair'
  | 'cleaning'
  | 'company'
  | 'emergency'
  | 'shopping'
  | 'transport'
  | 'rehab'
  | 'psychology'
  | 'legal_aid'
  | 'other'

// 全部服务类型（与后端 consts/service_types.go 一致）
export const allServiceTypes: ServiceType[] = [
  { key: 'meal', name: '助餐服务', icon: '🍱', description: '营养配餐送上门' },
  { key: 'medical', name: '就医陪护', icon: '🏥', description: '陪同就医看诊' },
  { key: 'care', name: '日常照护', icon: '👴', description: '生活起居照料' },
  { key: 'repair', name: '居家维修', icon: '🔧', description: '家电水电维修' },
  { key: 'cleaning', name: '家政保洁', icon: '🧹', description: '居家清洁服务' },
  { key: 'company', name: '陪伴聊天', icon: '💬', description: '陪伴交流慰藉' },
  { key: 'emergency', name: '紧急救助', icon: '🚨', description: '突发紧急情况' },
  { key: 'shopping', name: '代买代购', icon: '🛒', description: '生活用品代购' },
  { key: 'transport', name: '出行接送', icon: '🚗', description: '出行陪同接送' },
  { key: 'rehab', name: '康复理疗', icon: '💪', description: '康复训练指导' },
  { key: 'psychology', name: '心理慰藉', icon: '🧠', description: '心理咨询辅导' },
  { key: 'legal_aid', name: '法律援助', icon: '⚖️', description: '法律咨询服务' },
  { key: 'other', name: '其他服务', icon: '📋', description: '其他养老服务' }
]

// 获取服务类型名称
export function getServiceTypeName(key: string): string {
  const service = allServiceTypes.find(s => s.key === key)
  return service?.name || '未知服务'
}

// 获取服务类型图标
export function getServiceTypeIcon(key: string): string {
  const service = allServiceTypes.find(s => s.key === key)
  return service?.icon || '📋'
}
